packages/ml-vision/ios/RNFBMLVision/RNFBMLVisionTextRecognizerModule.m

```diff
@@ -0,0 +1,192 @@
+
+    FIRVisionImage *visionImage = [[FIRVisionImage alloc] initWithImage:image];
+    FIRVision *vision = [FIRVision vision];
+
+    FIRVisionCloudTextRecognizerOptions *options = [[FIRVisionCloudTextRecognizerOptions alloc] init];
+
+    if (cloudTextRecognizerOptions[@"hintedLanguages"]) {
+      options.languageHints = cloudTextRecognizerOptions[@"hintedLanguages"];
+    }
+
+    NSInteger *modelType = [cloudTextRecognizerOptions[@"modelType"] pointerValue];
+    if (modelType == (NSInteger *) 1) {
+      options.modelType = FIRVisionCloudTextModelTypeSparse;
+    } else if (modelType == (NSInteger *) 2) {
+      options.modelType = FIRVisionCloudTextModelTypeDense;
+    }
+
+    FIRVisionTextRecognizer *textRecognizer = [vision cloudTextRecognizerWithOptions:options];
+
+    [textRecognizer processImage:visionImage completion:^(FIRVisionText *text, NSError *error) {
+      if (error != nil) {
+        [RNFBSharedUtils rejectPromiseWithUserInfo:reject userInfo:(NSMutableDictionary *) @{
+            @"code": @"unknown",
+            @"message": [error localizedDescription],
+        }];
+        return;
+      }
+
+      resolve([self getFirebaseVisionTextMap:text]);
+    }];
+  }];
+}
+
+- (NSDictionary *)getFirebaseVisionTextMap:(FIRVisionText *)text {
+  NSMutableDictionary *firebaseVisionTextMap = [[NSMutableDictionary alloc] init];
+
+  firebaseVisionTextMap[@"text"] = text.text;
+  firebaseVisionTextMap[@"blocks"] = [self getVisionTextBlocksList:text.blocks];
+
+  return firebaseVisionTextMap;
+}
+
+- (NSArray *)getVisionTextBlocksList:(NSArray <FIRVisionTextBlock *> *)blocks {
+  NSMutableArray *blockListFormatted = [[NSMutableArray alloc] init];
+
+  for (FIRVisionTextBlock *block in blocks) {
+    NSMutableDictionary *textBlockFormatted = [[NSMutableDictionary alloc] init];
+
+    textBlockFormatted[@"text"] = block.text;
+    textBlockFormatted[@"confidence"] = block.confidence;
+    textBlockFormatted[@"recognizedLanguages"] = [self getLanguageCodesList:block.recognizedLanguages];
+//    textBlockFormatted[@"cornerPoints"] = [RNFBMLVisionCommon visionPointsToArray:block.cornerPoints];
+    textBlockFormatted[@"cornerPoints"] = @[];
+    textBlockFormatted[@"lines"] = [self getLinesList:block.lines];
+
+    [blockListFormatted addObject:textBlockFormatted];
+  }
+
+  return blockListFormatted;
+}
+
+- (NSArray *)getLinesList:(NSArray <FIRVisionTextLine *> *)lines {
+  NSMutableArray *lineListFormatted = [[NSMutableArray alloc] init];
+
+  for (FIRVisionTextLine *line in lines) {
+    NSMutableDictionary *lineFormatted = [[NSMutableDictionary alloc] init];
+
+    lineFormatted[@"boundingBox"] = [RNFBMLVisionCommon rectToIntArray:line.frame];
+    lineFormatted[@"text"] = line.text;
+    lineFormatted[@"confidence"] = line.confidence;
+    lineFormatted[@"recognizedLanguages"] = [self getLanguageCodesList:line.recognizedLanguages];
+//    lineFormatted[@"cornerPoints"] = [RNFBMLVisionCommon visionPointsToArray:line.cornerPoints];
```

Comment:
Corner points to point array issue

Reply:
Fixed in commit I just pushed from your Mac